qcodes/instrument/channel.py

```diff
@@ -0,0 +1,301 @@
+        self.functions = {}
+
+        self.name = "{}_{}".format(parent.name, str(name))
+        self._meta_attrs = ['name']
+
+        self._parent = parent
+
+    def __repr__(self):
+        """Custom repr to give parent information"""
+        return '<{}: {} of {}: {}>'.format(type(self).__name__, self.name,
+            type(self._parent).__name__, self._parent.name)
+
+    # We aren't a member of the global list of instruments, don't try and remove ourself
+    def __del__(self):
+        """ Does nothing for an instrument channel """
+        pass
+
+    def close(self):
+        """ Doesn't make sense to just close a channel by default, raise NotImplemented """
+        raise NotImplementedError("Can't close a channel. Close my parent instead.")
+
+    @classmethod
+    def record_instance(cls, instance):
+        """ Instances should not be recorded for channels. This should happen for the parent instrument. """
+        pass
+
+    @classmethod
+    def instances(cls):
+        """ Instances should not be recorded for channels. This should happen for the parent instrument. """
+        pass
+
+    @classmethod
+    def remove_instances(cls, instance):
+        """ It doesn't make sense to remove a channel from an instrument, raise NotImplemented"""
+        raise NotImplementedError("Can't remove a channel.")
+
+    # This method doesn't make sense for a channel, raise NotImplemented
+    @classmethod
+    def find_instruments(cls, name, instrument_class=None):
+        raise NotImplementedError("Can't find instruments in a channel")
+
+    # Pass any commands to read or write from the instrument up to the parent
+    def write(self, cmd):
+        return self._parent.write(cmd)
+    def write_raw(self, cmd):
+        return self._parent.write_raw(cmd)
+
+    def ask(self, cmd):
+        return self._parent.ask(cmd)
+    def ask_raw(self, cmd):
+        return self._parent.ask_raw(cmd)
+
+class ChannelList(Metadatable):
+    """
+    Container for channelized parameters that allows for sweeps over all channels, as well
+    as addressing of individual channels.
+
+    Args:
+        parent (Instrument): the instrument to which this channel should be attached
+
+        name (string): the name of the channel list
+
+        chan_type (InstrumentChannel): the type of channel contained within this list
+
+        chan_list (Iterable[chan_type]): An optional iterable of channels of type chan_type.
+            This will create a list and immediately lock the ChannelList.
+
+    Attributes:
+        parameters (Dict[Parameter]): All the parameters supported by this group of channels.
+
+        functions (Dict[Function]): All the functions supported by this group of channels
+    """
+
+    def __init__(self, parent, name, chan_type, chan_list=None):
+        super().__init__()
+
+        self._parent = parent
+        self._name = name
+        if not isinstance(chan_type, type) != type or not issubclass(chan_type, InstrumentChannel):
```

Comment:
As far as I can see the first part of this condition will always be True?

not isinstance(chan_type, type) != type -> not bool != type -> not False.